--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -154,7 +154,7 @@
  endif()
 
  # Installation : Final destination of the python modules
- set( TRIQS_PYTHON_LIB_DEST_ROOT "lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages" )
+ set( TRIQS_PYTHON_LIB_DEST_ROOT "lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/dist-packages" )
  set( TRIQS_PYTHON_LIB_DEST ${TRIQS_PYTHON_LIB_DEST_ROOT}/pytriqs )
  message(STATUS "Python modules will be installed in ${CMAKE_INSTALL_PREFIX}/${TRIQS_PYTHON_LIB_DEST}")
 
